src/Transients.cc

```diff
@@ -287,36 +296,35 @@ void
 Transients::abandon(const StoreEntry &e)
 {
     assert(e.mem_obj && map);
-    map->freeEntry(e.mem_obj->xitTable.index); // just marks the locked entry
-    CollapsedForwarding::Broadcast(e);
+    // avoid useless broadcasts
+    if (map->freeEntry(e.mem_obj->xitTable.index)); // just marks the locked entry
+        CollapsedForwarding::Broadcast(e, true);
     // We do not unlock the entry now because the problem is most likely with
     // the server resource rather than a specific cache writer, so we want to
     // prevent other readers from collapsing requests for that resource.
 }
 
-bool
-Transients::abandoned(const StoreEntry &e) const
-{
-    assert(e.mem_obj);
-    return abandonedAt(e.mem_obj->xitTable.index);
-}
-
-/// whether an in-transit entry at the index is now abandoned by its writer
-bool
-Transients::abandonedAt(const sfileno index) const
+void
+Transients::status(const StoreEntry &entry, bool &aborted, bool &waitingToBeFreed) const
 {
     assert(map);
-    return map->readableEntry(index).waitingToBeFreed;
+    assert(entry.mem_obj);
+    const auto idx = entry.mem_obj->xitTable.index;
+    const auto &anchor = collapsedWriter(entry) ?
+        map->writeableEntry(idx) : map->readableEntry(idx);
+
```

Comment:
I would remove this empty line: All the lines above and below it are about extracting those two flags.

Reply:
Still open. Please also remove the new empty line in Transients::copyFromShm().